### PR TITLE
SW-6135 Implemented admin UI functionality to migrate project details to variables

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2488** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2488?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2486** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2486?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2485** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2485?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/accelerator/model/ProjectAcceleratorDetailsModel.kt [hunk 1]
```diff
@@ -78,4 +78,26 @@ data class ProjectAcceleratorDetailsModel(
...
+          this.landUseModelTypes,
+          this.maxCarbonAccumulation,
+          this.minCarbonAccumulation,
+          this.numNativeSpecies,
+          this.perHectareBudget,
+          this.projectId,
+          this.region,
+          this.totalCarbon,
+          this.totalExpansionPotential,
+          this.whatNeedsToBeTrue,
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/accelerator/model/ProjectAcceleratorDetailsModel.kt]: No need for the `this.` prefixes here.

File: src/main/kotlin/com/terraformation/backend/admin/AdminAcceleratorProjectsController.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        acceleratorProjectVariableValuesService.writeValues(
+            acceleratorDetail.projectId, acceleratorDetail.toVariableValuesModel())
+        log.info("Variables migration for project ${acceleratorDetail.projectId} succeeded.")
+      } catch (e: Exception) {
+        problems.add("Project ${acceleratorDetail.projectId}: ${e.message}")
+        log.warn("Migration for project ${acceleratorDetail.projectId} failed", e)
+      }
+    }
+
+    if (problems.size == 0) {
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/admin/AdminAcceleratorProjectsController.kt]: Could be `problems.isEmpty()`

**Review by sgrimm — APPROVED** [review]

**Comment by tommylau523** [comment]
### Merge activity

* **Oct 15, 5:45 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2486).
* **Oct 15, 5:46 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2486) rebased this pull request as part of a merge.
* **Oct 15, 6:45 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2486).
* **Oct 15, 6:46 PM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2486).

**Pull request merged**